### PR TITLE
Fix version number parsing for newer kolibri-explore-plugin versions

### LINKED ISSUE #705: Release name of the Android app has changed
Screenshot from 2023-07-17 08-21-33

Instead of our release name like _Ladybird 6.26-365_ I see the date there _2023-07-13_

### PR DESCRIPTION
**Pull request by pwithnall**
How the version number is exported changed in kolibri-explore-plugin 6.27.0, so adapt the APK build code accordingly.

Fixes: https://github.com/endlessm/kolibri-explore-plugin/issues/705

### CODE REVIEW

**Comment by pwithnall** [comment]
Now I get the following:
```sh
$ make dist/version.json EXPLOREPLUGIN_WHEEL_PATH=./whl/kolibri_explore_plugin-6.28.0-py3-none-any.whl
…
echo '{"versionCode": "1689592200", "versionName": "Ladybird 6.28-1689592200", "ekVersion": "6.28.0-0.16.0.dev0-8f51254"}' > dist/version.json
```
(using the 6.28 `.whl` in the `whl/` directory)

**Comment by wjt** [comment]
Does this want a

> Fixes https://github.com/endlessm/kolibri-explore-plugin/issues/705

in the description?

**Comment by pwithnall** [comment]
> Does this want a
> 
> > Fixes [endlessm/kolibri-explore-plugin#705](https://github.com/endlessm/kolibri-explore-plugin/issues/705)
> 
> in the description?

Yes it most certainly does

**Comment by wjt** [comment]
I'm not entirely following what has changed in the way the version number is used, but I wonder if this is related to the problem described at https://github.com/endlessm/kolibri-explore-plugin/issues/703#issuecomment-1638011223 – the version number is part of the URLs used for the explore plugin JS and CSS and both those requests are 404-ing.

File: scripts/version.py [hunk 1]
```diff
@@ -32,6 +32,22 @@ def explore_plugin_version_name():
...
+            f"./{EXPLOREPLUGIN_TARGET}/kolibri_explore_plugin/_version.py", "r"
+        ) as version_file:
+            # The _version.py file always has the plugin version between quotes,
+            # in an assignment to __version__:
+            lines = version_file.readlines()
+            for line in lines:
+                if line.startswith("__version__ = "):
+                    return line.split("'")[1]
+    except FileNotFoundError:
+        pass
```
Comment (wjt) [inline scripts/version.py]: Just a drive-by comment: given this file is a Python module that defines a constant, and given this program reading it is a Python module, we could import it:

https://docs.python.org/3/library/importlib.html#importing-a-source-file-directly

File: scripts/version.py [hunk 2]
```diff
@@ -57,7 +57,7 @@ def explore_plugin_version():
 
 def explore_plugin_simple_version():
     full_version = explore_plugin_version()
-    major, minor, patch = full_version.split(".")
+    [major, minor, patch, *_] = full_version.split(".")
```
Comment (dbnicholson) [inline scripts/version.py:62]: This looks like it would be a lot nicer using the generated `version_tuple` in kolibri-explore-plugin, but this should be good enough.

File: scripts/version.py (hunk 1 above)
Comment (pwithnall) [reply scripts/version.py]: Done

**Review by dbnicholson — APPROVED** [review]
Looks good. I left a future possible improvement, but this should do the trick!
